### PR TITLE
Update ponos

### PR DESCRIPTION
**Pull request by anandkumarpatel**
- update ponos
- update rabbitmq file
- update deps
- fix test

### CODE REVIEW

**Comment by anandkumarpatel** [comment]
have to solve this:
Error: Channel closed by server: 404 (NOT-FOUND) with message "NOT_FOUND - no queue 'docker.listener.container.state.poll' in vhost '/'"

File: lib/rabbitmq.js [hunk 1]
```diff
@@ -2,7 +2,61 @@
...
+  }, {
+    name: 'docker.events-stream.disconnected',
+    jobSchema: Joi.object({
+      host: Joi.string().uri({ scheme: 'http' }).required(),
+      org: Joi.number().required()
+    })
+  }],
+  tasks: [{
+    name: 'docker.event.publish',
+    jobSchema: Joi.object({
```
Comment (podviaznikov) [inline lib/rabbitmq.js]: maybe for private services jobs we can reuse the schema definition? E.x. just ref https://github.com/CodeNow/docker-listener/blob/master/lib/workers/docker.event.publish.js#L15

File: lib/rabbitmq.js [hunk 2]
```diff
@@ -34,8 +88,7 @@ publisher.createConnectedJob = function (type, host, org) {
   // also host needs to have http://
   publisher.publishEvent(type + '.events-stream.connected', {
     host: 'http://' + host,
-    org: org,
-    tags: org
```
Comment (podviaznikov) [inline lib/rabbitmq.js]: not used anymore?

File: lib/workers/docker.event.publish.js [hunk 3]
```diff
@@ -77,6 +77,7 @@ DockerEventPublish._handlePublish = (event, log) => {
       } else if (DockerEventPublish._isBuildContainer(event)) {
         rabbitmq.publishTask('on-image-builder-container-create', event)
       }
+      rabbitmq.publishEvent('container.life-cycle.created', event)
```
Comment (podviaznikov) [inline lib/workers/docker.event.publish.js:89]: nice!

**Comment by podviaznikov** [comment]
this looks great.
One big feedback: maybe for private jobs(internal to DL) we should reuse schemas and not define them twice?

File: lib/rabbitmq.js (hunk 2 above)
Comment (anandkumarpatel) [reply lib/rabbitmq.js]: nope don't see it used anymore

Comment (podviaznikov) [reply lib/rabbitmq.js]: it is still used in sauron: https://github.com/CodeNow/sauron/blob/d9c83ba01581c0291843422be807204386f3b040/lib/workers/docker.events-stream.connected.js#L23.
I'm going to update sauron

**Comment by anandkumarpatel** [comment]
- [x] @anandkumarpatel

File: lib/rabbitmq.js [hunk 4]
```diff
@@ -2,7 +2,34 @@
...
+    name: 'docker.event.publish',
+    jobSchema: schemas.dockerEventPublish
+  }, {
+    name: 'swarm.events-stream.connect',
+    jobSchema: schemas.eventsStreamConnect
+  }, {
+    name: 'docker.events-stream.connect',
+    jobSchema: schemas.eventsStreamConnect
+  }, {
+    name: 'container.state.poll',
```
Comment (anandkumarpatel) [inline lib/rabbitmq.js]: should not be defined unless your publishing from here. should mock in test

Comment (podviaznikov) [reply lib/rabbitmq.js]: yeah, publishing from here. It's a task

**Comment by anandkumarpatel** [comment]
snoop-a-loop$ npm start

> snoop-a-loop@1.0.0 start /Users/anandkumarpatel/run/snoop-a-loop
> ./node_modules/mocha/bin/mocha -t 10000 -b index.js

  Cleanup
    ✓ should fetch the instances (775ms)
    ✓ should delete/destroy the non-repo container (200ms)
    ✓ should delete/destroy the repo container (223ms)
1. New Service Containers
   Creating Container
     ✓ should fetch all template containers (778ms)
     ✓ should copy the source instance (2024ms)
     ✓ should create the build (191ms)
     ✓ should build the build (1143ms)
     ✓ should create an instance (313ms)
   Working Container
     ✓ should have a dockerContainer (4534ms)
     ✓ should get logs for that container (92ms)
     ✓ should be succsefully built (1010ms)
     ✓ should have a working terminal (2020ms)
2. New Repository Containers
   Create A Container
     Github
       ✓ should create a github org
       ✓ should fetch a github branch (619ms)
       ✓ should fetch a github repo branch (572ms)
     Source Context
       ✓ should fetch the source context (167ms)
       ✓ should fetch the source context versions (177ms)
     Context & Context Versions
       ✓ should create a context (189ms)
       ✓ should create a context version (1376ms)
       ✓ should fetch the stack analysis (991ms)
       ✓ should copy the files (5224ms)
       ✓ should create an AppCodeVersion (2336ms)
     Builds & Instances
       ✓ should create a build for a context version (386ms)
       ✓ should build the build (1002ms)
       ✓ should create an instance (875ms)
   Working Container
     ✓ should have a dockerContainer (2521ms)
     ✓ should get logs for that container (116ms)
     ✓ should be successfully built (1008ms)
     ✓ should have a working terminal (1135ms)
3. New Repository Containers created using a mirrored docker file
   Create A Container
     Github
       ✓ should create a github org
       ✓ should fetch a github branch (954ms)
       ✓ should fetch a github repo branch (582ms)
     Source Context
       ✓ should fetch the source context (234ms)
       ✓ should fetch the source context versions (172ms)
     Context & Context Versions
       ✓ should create a context (206ms)
       ✓ should create a context version (753ms)
       ✓ should update the context version when the dockerfile is mirrored (396ms)
       ✓ should fetch the stack analysis (212ms)
       ✓ should copy the files (2913ms)
       ✓ should create an AppCodeVersion (1273ms)
     Builds & Instances
       ✓ should create a build for a context version (794ms)
       ✓ should build the build (254ms)
       ✓ should create an instance (582ms)
   Working Container
     ✓ should have a dockerContainer (4026ms)
     ✓ should get logs for that container (220ms)
     ✓ should be successfully built (1507ms)
     ✓ should have a working terminal (1162ms)
     ✓ should reflect the mirrored dockerfile configuration (1131ms)
4. Rebuild Repo Container
   Rebuilding without Cache
     ✓ should deep copy the build (496ms)
     ✓ should rebuild the instance without cache (915ms)
   Working Container
     ✓ should have a container (8034ms)
     ✓ should get logs for that container (666ms)
     ✓ should be succsefully built (504ms)
     ✓ should have a working terminal (492ms)
5. Github Webhooks
   Creating Branch Container
     ✓ should created a new branch (1371ms)
     ✓ should create a new instance with the branch name (7019ms)
   Working Container
     ✓ should have a container
     ✓ should get logs for that container (586ms)
     ✓ should be succsefully built (502ms)
     ✓ should have a working terminal (1514ms)
6. Isolation
   Create Container To Isolate
     Create A Container
       Github
         - should create a github org
         - should fetch a github branch
         - should fetch a github repo branch
       Source Context
         - should fetch the source context
         - should fetch the source context versions
       Context & Context Versions
         - should create a context
         - should create a context version
         - should fetch the stack analysis
         - should copy the files
         - should create an AppCodeVersion
       Builds & Instances
         - should create a build for a context version
         - should build the build
         - should create an instance
     Working Container
       - should have a dockerContainer
       - should get logs for that container
       - should be successfully built
       - should have a working terminal
   Create Isolation
   - should create the isolation
   - should create the instances for that isolation
     Isolated Service Container
   
   ```
   - should have a dockerContainer
   - should get logs for that container
   - should be successfully built
   - should have a working terminal
   ```
   
     Isolated Repo Container
       - should have a dockerContainer
       - should get logs for that container
       - should be successfully built
       - should have a working terminal
7. Container To Container DNS
   Repo Instance
     ✓ should connect to the container from the master branch (234ms)
     ✓ should connect to the container from the newly created branch (if not isolated) (468ms)
   - should connect to the isolated container from the isolated branch
     Service Instance
   - should connect to the master branch repo instance
   - should connect to the creaated branch repo instance
8. Navi URLs
   Repo Instance
     ✓ should access the main container (289ms)
     ✓ should access the branch container (392ms)
   Service Instance
   - should connect to the service instance
   
   64 passing (1m)
   31 pending

snoop-a-loop$

**Pull request merged**